### PR TITLE
Roles create 1: Roles Create API.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

Enable users with the right set of privileges to create roles.

This PR completes **1.** 
--
~~0. Add `color:string` attribute to roles model, `null` values aren't allowed and should default to the app primary HEX color.~~
~~1. Add `RolesController#create` to create a new role while providing its name.~~
> When permissions is made Roles just like `RoomMeetingOptions` should take the default
configuration on `:create` and can be updated later on.
2. Extend the UI and create a CreateRole modal that prompts  the user for a role name when clicking a button in the Roles page of the organization settings.
3. Extend/Add required mutations for integration with the `RolesController#create` API and
show proper feedbacks.
 
### User story [Manage Roles]:
---
1. A User with the right set of priviledges authenticates.
2. User goes to the Organasation settings.
3. User goes to the Roles page.
4. User can select a role and:
    - Delete it (Not all roles can be deleted).
[OR]
    - Edit color, name, permissions.
[OR]
    - Add new role while providing its name and maybe color.

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3572

**Comment by KH-Amir-TN** [comment]
# REBASED ON 71d16a5df7baae445d26be8b1943efef04be49aa

File: spec/controllers/admin/roles_controller_spec.rb [hunk 1]
```diff
@@ -36,19 +36,35 @@
         create(:role, name: 'I')
       end
 
-      it 'orders the roles list by column and direction DESC' do
+      it 'orders the recordings list by column and direction DESC' do
```
Comment (farhatahmad) [inline spec/controllers/admin/roles_controller_spec.rb:39]: ? why this change

Comment (KH-Amir-TN) [reply spec/controllers/admin/roles_controller_spec.rb:39]: It's roles not recordings xD
I copied the test from recordings spec but forgot to alter the test description.

Comment (KH-Amir-TN) [reply spec/controllers/admin/roles_controller_spec.rb:39]: Don't know probably mistyped during a rebase.
Will change it.

**Review by farhatahmad — APPROVED** [review]
